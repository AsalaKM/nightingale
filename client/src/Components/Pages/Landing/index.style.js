import styled from "styled-components";

export const LandingDiv = styled.div`
  background: linear-gradient(to top, #925de5, #cec8d8, #fff);
  min-height: 100vh;
  width: 100vw;
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 0;
`;

export const MainLogo = styled.img`
  width: 60%;
  margin: 0 auto;
  @media (min-width: 700px) {
    display: block;
    width: 20%;
    margin: 0 auto;
  }
`;

export const View = styled.div`
  width: 100%;
  padding: 5rem 0.5rem 5rem 0.5rem;
  min-height: 100vh;
  text-align: center;
  @media (min-width: 768px) {
    padding: 8rem 5rem 5rem 5rem;
  }
  ,
  a,
  /* // eslint-disable-next-line prettier/prettier */
  Link {
    display: block;
    color: #925de5;
    :hover {
      display: block;
      color: #925de5;
    }
    ,
    :active {
      display: block;
      color: #925de5;
    }
    ,
    :visited {
      display: block;
      color: #925de5;
    }
    ,
    :focus {
      display: block;
      color: #925de5;
    }
  }
`;

export const ViewParagraph = styled.p`
  text-align: center;
  margin: 1.5rem 0 1.5rem 0;
  color: #000;
`;
