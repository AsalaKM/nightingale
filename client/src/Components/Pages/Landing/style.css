.landing{
    background: linear-gradient(to top, #925DE5, #cec8d8, #fff);
    height: 100vh;
    text-align: center;
}