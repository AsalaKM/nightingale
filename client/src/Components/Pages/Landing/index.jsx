import React, { Component } from "react";
import { Link } from "react-router-dom";
import Button from "../../Common/Button";
import Img from "../../../assets/mainLogo.png";
import ImgTitle from "../../../assets/title.png";
import ImgLogo from "../../../assets/logo.png";
import { LandingDiv, MainLogo, View, ViewParagraph } from "./index.style";

class Landing extends Component {
  state = {
    pageOne: true,
    pageTwo: false,
    pageThree: false,
    pageFour: false
  };

  automaticMove = () => {
     setTimeout(() => {
      this.setState({ pageOne: false, pageTwo: true })}, 2000)
  }
  
  onClickOne = () => {
      this.setState({ pageOne: false, pageTwo: true });
  };

  onClickTwo = () => {
    this.setState({ pageTwo: false, pageThree: true });
  };
  onClickThree = () => {
    this.setState({ pageThree: false, pageFour: true });
  };
  goHome = () => {
    this.props.history.push(`/home`);
  };

  render() {
    const { pageOne, pageTwo, pageThree, pageFour } = this.state;

    

    return (
      <>
        {pageOne && (
          <LandingDiv onClick={this.onClickOne}>
            <MainLogo src={Img} alt="logo" />
            {this.automaticMove()}
          </LandingDiv>
        )}
        {pageTwo && (
          <View>
            <MainLogo src={ImgTitle} alt="logo" />
            <MainLogo src={ImgLogo} alt="logo" />
            <ViewParagraph>
              Hey! I'm Nightingale, a personal buddy to help you enjoy school.
              <span role="img" aria-label="emoji">
                ðŸ¤—
              </span>
            </ViewParagraph>
            <Button value="Get Started" onClick={this.onClickTwo} />
            <Link to="/login">Already have an account? Log in here</Link>
          </View>
        )}

        {pageThree && (
          <View>
            <MainLogo src={ImgTitle} alt="logo" />
            <MainLogo src={ImgLogo} alt="logo" />
            <ViewParagraph>
              Chat with me every day about your school experiences to clear your
              head and get the chance to reflect on what affects your mood.
            </ViewParagraph>
            <Button value="Sure thing" onClick={this.onClickThree} />
          </View>
        )}
        {pageFour && (
          <View>
            <MainLogo src={ImgTitle} alt="logo" />
            <MainLogo src={ImgLogo} alt="logo" />
            <ViewParagraph>
              Even get a daily breakdown of your mood and experiences so you can
              spot trends!
            </ViewParagraph>
            <Button value="Got it" onClick={this.goHome} />
          </View>
        )}
      </>
    );
  }
}
export default Landing;
