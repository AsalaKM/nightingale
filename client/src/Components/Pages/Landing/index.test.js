import React from "react";
import renderer from "react-test-renderer";
import Landing from "./index.jsx";

describe("Landing", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Landing />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
