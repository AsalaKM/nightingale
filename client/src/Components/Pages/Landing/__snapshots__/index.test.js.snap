// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing renders correctly 1`] = `
<div
  className="sc-bwzfXH dCmxRV"
  onClick={[Function]}
>
  <img
    alt="logo"
    className="sc-htpNat jpugCE"
    src="mainLogo.png"
  />
</div>
`;
